var/spack/repos/builtin/packages/exago/package.py

```diff
@@ -17,7 +17,9 @@ class Exago(CMakePackage, CudaPackage, ROCmPackage):
     git = "https://github.com/pnnl/ExaGO.git"
     maintainers("ryandanehy", "cameronrutherford", "pelesh")
 
-    version("1.5.1", tag="v1.5.1", submodules=True)
+    version(
+        "1.5.1", tag="v1.5.1", commit="84e9faf9d9dad8d851075eba26038338d90e6d3a", submodules=True
+    )
     version("1.5.0", commit="227f49573a28bdd234be5500b3733be78a958f15", submodules=True)
     version("1.4.1", commit="ea607c685444b5f345bfdc9a59c345f0f30adde2", submodules=True)
     version("1.4.0", commit="4f4c3fdb40b52ace2d6ba000e7f24b340ec8e886", submodules=True)
```

Comment:
```suggestion
    version(
        "1.5.1", tag="v1.5.1", commit="84e9faf9d9dad8d851075eba26038338d90e6d3a", submodules=True
    )
    version(
        "1.5.0", tag="v1.5.0", commit="227f49573a28bdd234be5500b3733be78a958f15", submodules=True
    )
    version(
        "1.4.1", tag="v1.4.1", commit="ea607c685444b5f345bfdc9a59c345f0f30adde2", submodules=True
    )
    version(
        "1.4.0", tag="v1.4.0", commit="4f4c3fdb40b52ace2d6ba000e7f24b340ec8e886", submodules=True
    )
```

@eugeneswalker noticed the commit hash was wrong for 1.5.1 in https://github.com/spack/spack/pull/40676, so not sure how/when that went awry...

All versions have a matching tag, and they are intended to be stable tags, so I think it makes sense too keep track of both for all exago versions. Would be good to fix this/apply to remaining versions in ExaGO 1.6.0 release https://github.com/spack/spack/pull/40614 @ryandanehy

Reply:
yeah, keeping track of the commit sha too is exactly there to guard against accidental/malicious retagging of commits.